react/features/aeternity/components/TipButton.js

```diff
@@ -0,0 +1,303 @@
+     */
+    isOpen: boolean,
+
+    /**
+     * Fiat currency.
+     */
+    currency: string,
+
+    /**
+     * AE value
+     */
+    value: string,
+
+    /**
+     * Any error
+     */
+    error: string,
+
+    /**
+     * Is show loading
+     */
+    showLoading: boolean,
+
+    /**
+     * Message for the author
+     */
+    message: string,
+};
+
+const URLS = {
+    SUPER: 'https://superhero.com',
+    RAENDOM: 'https://raendom-backend.z52da5wt.xyz'
+};
+const CONTRACT_ADDRESS = 'ct_2AfnEfCSZCTEkxL5Yoi4Yfq6fF7YapHRaFKDJK3THMXMBspp5z';
+
+const aeternity = {
+    contract: null,
+    async initTippingContractIfNeeded(): void {
+        if (!client) {
+            throw new Error('Init sdk first');
+        }
+        if (this.contract) {
+            return;
+        }
+
+        this.contract = await client.getContractInstance(TIPPING_INTERFACE, { contractAddress: CONTRACT_ADDRESS });
+    },
+    async tip(url, title, amount): Promise {
+        return this.initTippingContractIfNeeded().then(() => this.contract.methods.tip(url, title, { amount }));
+    },
+    util: {
+        aeToAtoms(ae) {
+            return (new BigNumber(ae)).times(new BigNumber(1000000000000000000));
+        }
+    }
+};
+
+/**
+ * Aeternity tip button react version.
+ */
+class TipButton extends Component<Props, State> {
+    /**
+     * Initializes a new TipButton instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isOpen: false,
+            currency: 'eur',
+            value: '',
+            message: `button host ${window.location.host} tip to ${this.props.account}`,
+            error: '',
+            showLoading: false,
+        };
+
+        this._changeCurrency = this._changeCurrency.bind(this);
+        this._onToggleTooltip = this._onToggleTooltip.bind(this);
+        this._tokensToCurrency = this._tokensToCurrency.bind(this);
+        this._onSendTip = this._onSendTip.bind(this);
+        this._onSendTipComment = this._onSendTipComment.bind(this);
+        this._onChangeValue = this._onChangeValue.bind(this);
+        this._onTipDeepLink = this._onTipDeepLink.bind(this);
+    }
+
+    /**
+     * WIP.
+     * Chane currency.
+     *
+     * @param {string} currency - New currency.
+     * @returns {void}
+     */
+    _changeCurrency(currency) {
+        this.setState({ currency });
+    }
+
+    /**
+     * Toggle tooltip.
+     *
+     * @param {string} currency - New currency.
+     * @returns {void}
+     */
+    _onToggleTooltip() {
+        this.setState({ isOpen: !this.state.isOpen });
+    }
+
+    /**
+     * Change ae value.
+     *
+     * @param {Object} event - Contains new value.
+     * @returns {void}
+     */
+    _onChangeValue({ target: { value } }) {
+        this.setState({ value });
+    }
+
+    /**
+     * WIP.
```

Comment:
Please can you remove the WIP documentation in this place and other places

Reply:
OK